### PR TITLE
[CN-1273] Support IAM roles for service accounts in AWS

### PR DESCRIPTION
**Pull request by semihbkgr**
## Description

supporting IAM roles for service accounts in AWS
see: #983

## User Impact

AWS IAM roles for service accounts are supported in bucket operations

### CODE REVIEW

File: config/crd/bases/hazelcast.com_hazelcasts.yaml [hunk 1]
```diff
@@ -146,7 +146,7 @@ spec:
                         type: object
                     type: object
                   version:
-                    default: 0.1.27
+                    default: CN-1273-3
```
Comment (hasancelik) [inline config/crd/bases/hazelcast.com_hazelcasts.yaml:149]: will be updated

File: helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml [hunk 2]
```diff
@@ -525,7 +525,7 @@ spec:
                         type: object
                     type: object
                   version:
-                    default: 0.1.27
+                    default: CN-1273-3
```
Comment (hasancelik) [inline helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml:528]: will be updated

File: api/v1alpha1/hazelcast_types.go [hunk 3]
```diff
@@ -679,12 +679,12 @@ func (c *UserCodeDeploymentConfig) IsRemoteURLsEnabled() bool {
...
 type AgentConfiguration struct {
 	// Repository to pull Hazelcast Platform Operator Agent(https://github.com/hazelcast/platform-operator-agent)
-	// +kubebuilder:default:="docker.io/hazelcast/platform-operator-agent"
+	// +kubebuilder:default:="docker.io/semihbkgr/hazelcast-platform-operator-agent"
 	// +optional
 	Repository string `json:"repository,omitempty"`
 
 	// Version of Hazelcast Platform Operator Agent.
-	// +kubebuilder:default:="0.1.27"
+	// +kubebuilder:default:="CN-1273-3"
```
Comment (hasancelik) [inline api/v1alpha1/hazelcast_types.go:687]: will be updated

File: helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml [hunk 4]
```diff
@@ -472,7 +472,7 @@ spec:
                 description: B&R Agent configurations
                 properties:
                   repository:
-                    default: docker.io/hazelcast/platform-operator-agent
+                    default: docker.io/semihbkgr/hazelcast-platform-operator-agent
```
Comment (hasancelik) [inline helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml:475]: will be updated

File: test/e2e/backup_restore_test.go [hunk 5]
```diff
@@ -584,6 +584,7 @@ var _ = Describe("Hazelcast CR with Persistence feature enabled", Group("backup_
 			backupRestore(hazelcast, hotBackup, false)
 		},
 			Entry("using GCP Workload Identity", Tag(EE|GCP), "cn-workload-identity-test", "gs://operator-e2e-external-backup"),
+			Entry("using AWS IAM ServiceAccount", Tag(EE|AWS), "aws-iam-sa", "s3://operator-e2e-external-backup"),
```
Comment (hasancelik) [inline test/e2e/backup_restore_test.go:587]: 👍

**Review by hasancelik — CHANGES_REQUESTED** [review]

File: config/crd/bases/hazelcast.com_hazelcasts.yaml (hunk 1 above)
Comment (semihbkgr) [reply config/crd/bases/hazelcast.com_hazelcasts.yaml:149]: Thanks for the heads up.
I will release new version of agent first, then update the default version here.

File: helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml (hunk 4 above)
Comment (semihbkgr) [reply helm-charts/hazelcast-platform-operator/charts/hazelcast-platform-operator-crds/templates/all-crds.yaml:475]: 👍 It will be reverted

**Review by hasancelik — APPROVED** [review]

**Review by SeriyBg — APPROVED** [review]

**Pull request merged**